hi-ml-cpath/testhisto/testhisto/models/test_deepmil.py

```diff
@@ -425,3 +427,49 @@ def test_class_weights_multiclass() -> None:
     # TODO: the test should reflect actual weighted loss operation for the class weights after
     # batch_size > 1 is implemented.
     assert allclose(loss_weighted, loss_unweighted)
+
+
+@pytest.mark.parametrize("tune_encoder", [True, False])
+@pytest.mark.parametrize("tune_pooling", [True, False])
```

Comment:
If it's possible to backprop when `tune_encoder=True` and `tune_pooling=False`, why can we not also have a tuning flag for the classifier? What's different?

Reply:
Still getting the same error with the gradients if I exclude the classifier from the optimizer or disable its gradients.

I thought that it was possible if we exclude it from the optimizer but keep its gradients enabled. While debugging step by step, it seemed to work, but it turned out that at the end of the training epoch it still fails with the same error I shared yesterday.

I am not sure about this, but my intuition is that we MUST start the backpropagation from the loss function where we evaluate the error (y_pred vs y_true). It's still okay to backprop if some intermediate layers are frozen but if we exclude the classifier there is no way for the optimizer to derive the gradients while being disconnected from the loss/error evaluation?

Does it make sense?